src/global_distrib/mod_global_distrib_mapping.erl

```diff
@@ -0,0 +1,290 @@
+%%==============================================================================
+%% Copyright 2017 Erlang Solutions Ltd.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%==============================================================================
+
+-module(mod_global_distrib_mapping).
```

Comment:
I think the task of going through all exported functions and documenting them properly (probably with a spec, too) should be made into a separate task so that we can merge the actual changes right now and stop the world conflicts.

Reply:
That's totally fine, let's just make sure we're planning on doing this :)